pkg/network/ebpf/c/tracer-stats.h

```diff
@@ -0,0 +1,105 @@
+#ifndef __TRACER_STATS_H
+#define __TRACER_STATS_H
+
+#include "tracer.h"
+
+static int read_conn_tuple(conn_tuple_t* t, struct sock* skp, u64 pid_gid, metadata_mask_t type);
+
+static __always_inline void update_conn_state(conn_tuple_t* t, conn_stats_ts_t *stats, size_t sent_bytes, size_t recv_bytes) {
+    if (t->metadata&CONN_TYPE_TCP || stats->flags&CONN_ASSURED) {
+        return;
+    }
+
+    if (stats->recv_bytes == 0 && sent_bytes > 0) {
+        stats->flags |= CONN_L_INIT;
+        return;
+    }
+
+    if (stats->sent_bytes == 0 && recv_bytes > 0) {
+        stats->flags |= CONN_R_INIT;
+        return;
+    }
+
+    // If a three-way "handshake" was established, we mark the connection as assured
+    if ((stats->flags&CONN_L_INIT && stats->recv_bytes > 0 && sent_bytes > 0)
+        || (stats->flags&CONN_R_INIT && stats->sent_bytes > 0 && recv_bytes > 0)) {
+        stats->flags |= CONN_ASSURED;
```

Comment:
Does CONN_ASSURED need to be its own flag? Could we check that `flags&(CONN_L_INIT|CONN_R_INIT) > 0` in userspace?

Reply:
The sent and recv bytes checks also play a role, so we need that extra flag.